### PR TITLE
[bugfix/MOB-273] Search Store Apps fix

### PR DESCRIPTION
**Pull request by franciscoaleixo**
**What does this PR do?**

   This fixes an issue where if searching an app in a specific store resulted in other results (like from apps & catappult store).

**Database changed?**

   No

**How should this be manually tested?**

  Go to stores and follow 'nzxt', then go to the store page and search for an app (e.g. 'tft'), it should only return apps inside that store, and not others.

**What are the relevant tickets?**

  [MOB-273](https://aptoide.atlassian.net/browse/MOB-273)

**Code Review Checklist**

- [ ] Documentation on public interfaces
- [ ] Database changed? If yes - Migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] New Kotlin code has unit tests
- [ ] New flows in presenters unit tests
- [ ] Mappers/Validators with any kind of logic unit tests
- [ ] Functional tests pass

### CODE REVIEW

File: app/src/main/java/cm/aptoide/pt/search/view/SearchResultPresenter.java [hunk 1]
```diff
@@ -205,15 +205,24 @@ private int getItemCount(List<SearchAppResult> data) {
...
         .doOnNext(__ -> view.showLoadingMore())
-        .flatMapSingle(viewModel -> loadDataFromFollowedStores(viewModel.getSearchQueryModel()
-            .getFinalQuery(), viewModel.isOnlyTrustedApps(), viewModel.getFollowedStoresOffset()))
+        .flatMapSingle(viewModel -> {
+          String storeName = viewModel.getStoreName();
+          if (storeName != null && !storeName.trim()
+              .equals("")) {
+            return loadDataForSpecificStore(viewModel.getSearchQueryModel()
+                    .getFinalQuery(), storeName, viewModel.getFollowedStoresOffset()).map(
+                    searchResult -> new Pair<>(searchResult, null));
```
Comment (phillipsgoncalves) [inline app/src/main/java/cm/aptoide/pt/search/view/SearchResultPresenter.java:214]: why did you add this map for this pair ?
This is the reason why you have to cast data to searchresult on line #225

Comment (franciscoaleixo) [reply app/src/main/java/cm/aptoide/pt/search/view/SearchResultPresenter.java:214]: No idea why I did that 🚀
